src/onnx/parse_if.cpp

```diff
@@ -0,0 +1,63 @@
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/onnx/onnx_parser.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/make_op.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_if : op_parser<parse_if>
+{
+    std::vector<op_desc> operators() const { return {{"If"}}; }
+
+    std::vector<instruction_ref> parse(const op_desc& /*opd*/,
+                                       onnx_parser& parser,
+                                       const onnx_parser::node_info& info,
+                                       std::vector<instruction_ref> args) const
+    {
+        migraphx::argument cond_arg = args.front()->eval();
+        // cond is not constant, need to create sub_modules
+        if(cond_arg.empty())
+        {
+            MIGRAPHX_THROW(
+                "PARSE_IF: current implementation requires condition input to be constant!");
+        }
+
+        std::vector<bool> vec_conds;
+        cond_arg.visit([&](auto s) { vec_conds.assign(s.begin(), s.end()); });
```

Comment:
You should just do `cond_arg.at<bool>()` to get a boolean.

Reply:
fixed